core/src/main/java/jenkins/security/OverriddenAuthorizationStrategy.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 IKEDA Yasuyuki
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package jenkins.security;
+
+import java.util.Collection;
+
+import javax.annotation.Nonnull;
+
+import org.acegisecurity.Authentication;
+
+import hudson.model.AbstractItem;
+import hudson.model.AbstractProject;
+import hudson.model.Computer;
+import hudson.model.Job;
+import hudson.model.Node;
+import hudson.model.User;
+import hudson.model.View;
+import hudson.security.ACL;
+import hudson.security.AuthorizationStrategy;
+import hudson.security.Permission;
+import hudson.slaves.Cloud;
+import jenkins.model.Jenkins;
+
+/**
+ * {@link AuthorizationStrategy} overriding another {@link AuthorizationStrategy}
+ * with {@link AuthorizationStrategyOverrideBase}.
+ *
+ * @since TODO
+ */
+public class OverriddenAuthorizationStrategy extends AuthorizationStrategy {
+    private final AuthorizationStrategy baseStrategy;
+
+    /**
+     * Creates a new instance wrapping another {@link AuthorizationStrategy}.
+     *
+     * @param baseStrategy a strategy to override.
+     */
+    public OverriddenAuthorizationStrategy(@Nonnull AuthorizationStrategy baseStrategy) {
+        this.baseStrategy = baseStrategy;
+    }
+
+    /**
+     * @return wrapped {@link AuthorizationStrategy}
+     */
+    @Nonnull
+    public AuthorizationStrategy getBaseStrategy() {
+        return baseStrategy;
+    }
+
+    @Nonnull
+    private <T> ACL overrideACL(@Nonnull final ACL baseAcl, @Nonnull final T item, @Nonnull final Class<T> klazz) {
+        return new ACL() {
+            @Override
+            public boolean hasPermission(@Nonnull Authentication a, @Nonnull Permission p) {
```

Comment:
This may reduce the performance issue:
```
  final List<> overrides = AuthorizationStrategyOverrideConfiguration.enables(klazz);
  if (overrides.isEmpty()) {
    return baseAcl;
  }
  return new ACL() {
    ...
      for (AuthorizationStrategyOverrideBase d: overrides) {
        ...
      }
    ...
  }
}
```